### PR TITLE
Updated README primarily to cover modes of operation.

### PR DESCRIPTION
**Pull request by dgoodwin**

### CODE REVIEW

**Comment by joelddiaz** [comment]
I haven't yet reviewed this, but as a heads up, there is another outstanding PR that covers part of this openshift/cloud-credential-operator#188

File: README.md [hunk 1]
```diff
@@ -26,11 +26,114 @@ NOTE: at present, only some components have transitioned to this new model, the
...
+OpenStack | kube-system/openstack-credentials | clouds.yaml
+VSphere | kube-system/vsphere-creds |
+oVirt | kube-system/ovirt-credentials | ovirt_url ovirt_username ovirt_password ovirt_cafile ovirt_insecure ovirt_ca_bundle
+
+
+## Modes of Operation
+
+### 1. Mint Mode
+
+The default and recommended best practice for running OpenShift is to run the installer with an admin level cloud credential. The admin credential is stored in kube-system namespace and then used by the cloud credential operator to process the CredentialRequests in the cluster and create additional fine grained permissions for each OpenShift component containing only the permissions it needs.
```
Comment (joelddiaz) [inline README.md:49]:
```suggestion
The default and recommended best practice for running OpenShift is to run the installer with an admin level cloud credential. The admin credential is stored in kube-system namespace and then used by the cloud credential operator to process the CredentialRequests in the cluster and create new users for each CredentialsRequest with additional fine grained permissions for each OpenShift component containing only the permissions it needs.
```

File: README.md [hunk 2]
```diff
@@ -26,11 +26,114 @@ NOTE: at present, only some components have transitioned to this new model, the
...
+
+Pros:
+  * Each cluster component has only the permissions it needs.
+  * Automatic on-going reconciliation for cloud credentials including upgrades.
```
Comment (joelddiaz) [inline README.md:53]:
```suggestion
  * Automatic on-going reconciliation for cloud credentials including upgrades (that require new/additional permissions).
```

File: README.md [hunk 3]
```diff
@@ -26,11 +26,114 @@ NOTE: at present, only some components have transitioned to this new model, the
...
+Pros:
+  * Admin credential is not stored in the cluster permanently and does not need to be long lived.
+Cons:
+  * Still requires admin credential in the cluster for brief periods of time.
+  * Requires manually re-instating the Secret with admin credentials for each upgrade.
+
+
+### 2. Passthrough Mode
+
+In this mode a user installs OpenShift with a single credential that *is not* an admin and cannot mint additional credentials, but itself has enough permissions to perform all operations needed by all components in the cluster. The cloud credential operator then shares this credential to each component.
```
Comment (joelddiaz) [inline README.md:78]:
```suggestion
In this mode a user installs OpenShift with a single credential that *is not* an admin and cannot mint additional credentials, but itself has enough permissions to perform the installation and all operations needed by all components in the cluster. The cloud credential operator then shares this credential to each component.
```

File: README.md [hunk 4]
```diff
@@ -26,11 +26,114 @@ NOTE: at present, only some components have transitioned to this new model, the
...
+Cons:
+  * Includes broad permissions only needed at install time, unless manual action is taken to reduce permissions post-install.
+  * Credential permissions may need to be manually updated prior to any upgrade.
+  * Each component has permissions used by all other components.
+
+Supported clouds: AWS, GCP, Azure, VMWare, OpenStack, oVirt
+
+### 3. Manual Credentials Management
+
+In this mode a user manually performs the job of the cloud credential operator. This requires examinings the CredentialsRequests in an OpenShift 4 release image, creating credentials in the underlying cloud provider, and finally creating Kubernetes Secrets in the correct namespaces to satisfy
```
Comment (joelddiaz) [inline README.md:101]:
```suggestion
In this mode a user manually performs the job of the cloud credential operator. This requires examining the CredentialsRequests in an OpenShift 4 release image, creating credentials in the underlying cloud provider, and finally creating Kubernetes Secrets in the correct namespaces to satisfy all CredentialsRequests for the current cloud.
```

**Comment by dgoodwin** [comment]
Updated with review comments, new content, and WIP removed.

**Pull request merged**